site/src/pages/TemplatePage/TemplatePageView.tsx

```diff
@@ -36,104 +39,141 @@ export interface TemplatePageViewProps {
+    <Link underline="none" component={RouterLink} to={`/templates/${template.name}/workspace`}>
+      <Button className={className} startIcon={<AddCircleOutline />}>
+        {Language.createButton}
+      </Button>
+    </Link>
+  )
+
   return (
     <Margins>
-      <PageHeader
-        actions={
-          <Stack direction="row" spacing={1}>
-            <Link
-              underline="none"
-              component={RouterLink}
-              to={`/templates/${template.name}/settings`}
-            >
-              <Button variant="outlined" startIcon={<SettingsOutlined />}>
-                {Language.settingsButton}
-              </Button>
-            </Link>
-            <Link
-              underline="none"
-              component={RouterLink}
-              to={`/templates/${template.name}/workspace`}
-            >
-              <Button startIcon={<AddCircleOutline />}>{Language.createButton}</Button>
-            </Link>
-          </Stack>
-        }
-      >
-        <Stack direction="row" spacing={3} className={styles.pageTitle}>
-          <div>
-            {hasIcon ? (
-              <div className={styles.iconWrapper}>
-                <img src={template.icon} alt="" />
-              </div>
-            ) : (
-              <Avatar className={styles.avatar}>{firstLetter(template.name)}</Avatar>
-            )}
-          </div>
-          <div>
-            <PageHeaderTitle>{template.name}</PageHeaderTitle>
-            <PageHeaderSubtitle>
-              {template.description === "" ? Language.noDescription : template.description}
-            </PageHeaderSubtitle>
-          </div>
-        </Stack>
-      </PageHeader>
+      <>
+        <PageHeader
+          actions={
+            <Stack direction="row" spacing={1}>
+              <Link
+                underline="none"
+                component={RouterLink}
+                to={`/templates/${template.name}/settings`}
+              >
+                <Button variant="outlined" startIcon={<SettingsOutlined />}>
+                  {Language.settingsButton}
+                </Button>
+              </Link>
 
-      <Stack spacing={2.5}>
-        <TemplateStats template={template} activeVersion={activeTemplateVersion} />
-        <WorkspaceSection
-          title={Language.resourcesTitle}
-          contentsProps={{ className: styles.resourcesTableContents }}
+              {canDeleteTemplate ? (
+                <DropdownButton
+                  primaryAction={createWorkspaceButton(styles.actionButton)}
+                  secondaryActions={[
+                    {
+                      action: "delete",
+                      button: (
+                        <DeleteButton handleAction={() => handleDeleteTemplate(template.id)} />
+                      ),
+                    },
+                  ]}
+                  canCancel={false}
+                />
+              ) : (
+                createWorkspaceButton("")
```

Comment:
We could change the arg to be optional, it is not intuitive what an empty string means here.